test/funnels/Depositor.t.sol

```diff
@@ -0,0 +1,485 @@
+        emit Withdraw(FACILITATOR, DAI, USDC, liq, 0, 0);
+        vm.prank(FACILITATOR); (uint256 withdrawn0, uint256 withdrawn1) = depositor.withdraw(wp);
+
+        assertTrue(
+            (withdrawn0 > deposited0 && GemLike(DAI ).balanceOf(address(buffer)) > initialDAI ) || 
+            (withdrawn1 > deposited1 && GemLike(USDC).balanceOf(address(buffer)) > initialUSDC)
+        );
+        assertTrue(GemLike(DAI).balanceOf(address(buffer)) >= prevDAI);
+        assertTrue(GemLike(USDC).balanceOf(address(buffer)) >= prevUSDC);
+        assertEq(GemLike(DAI).balanceOf(address(depositor)), 0);
+        assertEq(GemLike(USDC).balanceOf(address(depositor)), 0);
+        assertEq(NftLike(UNIV3_POS_MGR).balanceOf(address(buffer)), 1);        
+    }
+
+    function testMoveLiquidity() public {
+        Depositor.DepositParams memory dp = Depositor.DepositParams({ 
+            gem0: DAI,
+            gem1: USDC,
+            amt0: 500 * WAD, 
+            amt1: 500 * 10**6, 
+            minAmt0: 490 * WAD, 
+            minAmt1: 490 * 10**6, 
+            fee: uint24(100), 
+            tickLower: REF_TICK-100, 
+            tickUpper: REF_TICK+100
+        });
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+        bytes32 key = keccak256(abi.encode(dp.gem0, dp.gem1, dp.fee, dp.tickLower, dp.tickUpper));
+        uint256 tokenId = depositor.tokenIds(key);
+        address newBuffer = address(new AllocatorBuffer(ilk));
+        buffer.setApprovalForAll(UNIV3_POS_MGR, address(this), true);
+        assertEq(NftLike(UNIV3_POS_MGR).balanceOf(address(buffer)), 1);
+        assertEq(NftLike(UNIV3_POS_MGR).balanceOf(newBuffer), 0);
+
+        NftLike(UNIV3_POS_MGR).transferFrom(address(buffer), newBuffer, tokenId);
+
+        assertEq(NftLike(UNIV3_POS_MGR).balanceOf(address(buffer)), 0);
+        assertEq(NftLike(UNIV3_POS_MGR).balanceOf(newBuffer), 1);
+    }
+
+    function testDepositWrongGemOrder() public {
+        Depositor.DepositParams memory dp = Depositor.DepositParams({ 
+            gem0: USDC,
+            gem1: DAI,
+            amt0: 1, 
+            amt1: 1, 
+            minAmt0: 0, 
+            minAmt1: 0, 
+            fee: uint24(100), 
+            tickLower: REF_TICK-100, 
+            tickUpper: REF_TICK+100
+        });
+        vm.expectRevert("Depositor/wrong-gem-order");
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+    }
+
+    function testDepositTooSoon() public {
+        Depositor.DepositParams memory dp = Depositor.DepositParams({ 
+            gem0: DAI,
+            gem1: USDC,
+            amt0: 500 * WAD, 
+            amt1: 500 * 10**6, 
+            minAmt0: 490 * WAD, 
+            minAmt1: 490 * 10**6, 
+            fee: uint24(100), 
+            tickLower: REF_TICK-100, 
+            tickUpper: REF_TICK+100
+        });
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+        vm.expectRevert("Depositor/too-soon");
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+    }
+
```

Comment:
Missing a test for increasing liquidity after the position was already created (the second path in the code).

Reply:
Ah seems `testDeposit()` covers that, resolving.